influxdb3/src/commands/serve.rs

```diff
@@ -293,6 +296,25 @@ pub struct Config {
         action
     )]
     pub meta_cache_eviction_interval: humantime::Duration,
+
+    /// Threshold for internal buffer, can be either percentage or absolute value.
+    /// eg: 70% or 100000
+    #[clap(
+        long = "force-snapshot-mem-threshold",
+        env = "INFLUXDB3_FORCE_SNAPSHOT_MEM_THRESHOLD",
+        default_value = "70%",
+        action
+    )]
+    pub force_snapshot_mem_threshold: MemorySize,
+
+    /// Interval to check buffer size (and compare with `force_snapshot_mem_threshold`)
+    #[clap(
+        long = "force-snapshot-interval",
+        env = "INFLUXDB3_FORCE_SNAPSHOT_INTERVAL",
```

Comment:
I don't think this needs to be a configuration option. Just have it as a constant. You can then either initialize with that or, in tests, with something smaller.